Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/SharedViewBookmarkComponent.cpp

```diff
@@ -0,0 +1,116 @@
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<EditorViewBookmarks>("EditorViewBookmarks", "")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "Editor View Bookmarks")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &EditorViewBookmarks::m_viewBookmarks, "View Bookmarks", "")
+                    ->Attribute(AZ::Edit::Attributes::ContainerCanBeModified, true)
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->Attribute(AZ::Edit::Attributes::IndexedChildNameLabelOverride, &EditorViewBookmarks::GetBookmarkLabel);
+            }
+        }
+    }
+
+    AZStd::string EditorViewBookmarks::GetBookmarkLabel(int index) const
+    {
+        return AZStd::string::format("View Bookmark %d", index);
+    }
+
+    void SharedViewBookmarkComponent::Reflect(AZ::ReflectContext* context)
+    {
+        EditorViewBookmarks::Reflect(context);
+
+        if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->RegisterGenericType<EditorViewBookmarks>();
+
+            serializeContext->Class<SharedViewBookmarkComponent, EditorComponentBase>()->Version(0)->Field(
+                "ViewBookmarks", &SharedViewBookmarkComponent::m_viewBookmark);
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext
+                    ->Class<SharedViewBookmarkComponent>(
+                        "Shared View Bookmark Component", "The ViewBookmark Component allows to store bookmarks for a prefab")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::AddableByUser, true)
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Level", 0x9aeacc13))
+                    ->Attribute(AZ::Edit::Attributes::Category, "View Bookmarks")
+                    ->Attribute(AZ::Edit::Attributes::Icon, "Icons/Components/Comment.svg")
+                    ->Attribute(AZ::Edit::Attributes::ViewportIcon, "Icons/Components/Viewport/Comment.svg")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &SharedViewBookmarkComponent::m_viewBookmark, "ViewBookmarks", "ViewBookmarks")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, false);
+            }
+        }
+    }
+
+    AZStd::optional<ViewBookmark> SharedViewBookmarkComponent::GetBookmarkAtIndex(int index) const
+    {
+        if (index >= 0 && index < m_viewBookmark.m_viewBookmarks.size())
+        {
+            return AZStd::optional<ViewBookmark>(m_viewBookmark.m_viewBookmarks[index]);
+        }
+        return AZStd::optional<ViewBookmark>();
```

Comment:
I saw you went for option B. That's fine too.

Reply:
Yes I went to option 2 since a `ViewBookmark` with `m_position = Vector3::CreateZero()` and `m_rotation = Vector3::CreateZero()` although unlikely are still possible so that one should be a valid bookmark